src/button/buttonview.js

```diff
@@ -51,6 +53,25 @@ export default class ButtonView extends View {
 				}
 			}
 		};
+
+		this.register( 'children', el => el );
+	}
+
+	init() {
+		if ( this.model.icon ) {
+			this._setupIcon();
+		}
+
+		return super.init();
+	}
+
+	_setupIcon() {
+		const iconViewModel = new Model();
+		const iconView = new IconView( iconViewModel );
```

Comment:
Since you create view without creating the controller, the view won't be destroyed. If it has some listener for instance, it will not be detached.

That's why I think we need a simple method for creating adding one view inside another view. This can be done as a followup of course – nothing critical at this stage.